src/queries/SearchResultQuery.ts

```diff
@@ -26,42 +32,194 @@ class SearchResultQuery extends BaseQuery<SearchResult> implements SearchResultQ
+      case SearchType.EXACT:
+        return this.filterByExactKeywordAndUrl(queryInput);
+
+      case SearchType.PARTIAL:
+        return this.filterByPartialKeywordAndUrl(queryInput, leastCount);
+
+      default:
+        return this.filterByPatternKeywordAndUrl(queryInput);
+    }
+  }
 
-    this.filterQuery['OR'] = [
-      {
-        adWordTopUrls: {
-          has: this.filters?.url,
-        },
+  private async filterByExactUrl(queryInput: string): Promise<SearchResult[]> {
+    return this.databaseModel.findMany({
+      where: {
+        userId: this.userId,
+        OR: [
+          {
+            adWordTopUrls: {
+              has: queryInput,
+            },
+          },
+          {
+            nonAdUrls: {
+              has: queryInput,
+            },
+          },
+        ],
       },
-      {
-        nonAdUrls: {
-          has: this.filters?.url,
-        },
+      ...this.paginationQuery,
+    });
+  }
+
+  private async filterByExactKeywordAndUrl(queryInput: string): Promise<SearchResult[]> {
+    return this.databaseModel.findMany({
+      where: {
+        userId: this.userId,
+        keyword: queryInput,
+        OR: [
+          {
+            adWordTopUrls: {
+              has: queryInput,
+            },
+          },
+          {
+            nonAdUrls: {
+              has: queryInput,
+            },
+          },
+        ],
       },
-    ];
+      ...this.paginationQuery,
+    });
+  }
+
+  private async filterByPatternUrl(queryInput: string): Promise<SearchResult[]> {
+    return dbClient.$queryRaw(
+      Prisma.sql`SELECT * FROM search_results sr
+       WHERE sr."userId" = ${this.userId}
+       AND array_to_string(sr.ad_word_top_urls, ', ') ~ ${queryInput}
+       LIMIT ${this.paginationQuery.take} OFFSET ${this.paginationQuery.skip}`
+    );
+  }
+
+  private async filterByPatternKeywordAndUrl(queryInput: string): Promise<SearchResult[]> {
+    return dbClient.$queryRaw(
+      Prisma.sql`SELECT * FROM search_results sr
+       WHERE sr."userId" = ${this.userId}
+       AND sr."keyword" ~ ${queryInput}
+       AND array_to_string(sr.ad_word_top_urls, ', ') ~ ${queryInput}
+       LIMIT ${this.paginationQuery.take} OFFSET ${this.paginationQuery.skip}`
+    );
+  }
+
+  private async filterByPartialUrl(
+    queryInput: string,
+    leastCharacterCount: number
+  ): Promise<SearchResult[]> {
+    const characterToSearch = `%${queryInput}`.repeat(leastCharacterCount).concat('%');
+
+    return dbClient.$queryRaw(
+      Prisma.sql`
+      SELECT flatten.id, flatten.keyword, flatten.nod_ad_urls, flatten.ad_word_top_urls, flatten.status  FROM 
+        (SELECT sr.id,
+          sr.keyword,
+          UNNEST(sr.nod_ad_urls) as ad_url_string, 
+          UNNEST(sr.ad_word_top_urls) as non_ads_url_string, 
+          sr.nod_ad_urls, 
+          sr.ad_word_top_urls,
+          sr.status  
+          FROM search_results sr
+          WHERE sr."userId"=${this.userId}
+        ) AS flatten
+      where
```

Comment:
Casing inconsistency:

```suggestion
      WHERE
```